test_scripts/API/Restructuring_OnResetTimeout/023_BC_OnResetTimeout_SubtleAlert_with_softButton.lua

```diff
@@ -0,0 +1,77 @@
+------------------------------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md
+------------------------------------------------------------------------------------------------------------------------
```

Comment:
Please point me to the specifications for this. I believe onResetTimout should be applied to all the HMI interfacing RPCs, unless it is specified otherwise as per some other proposal or requirement. So if you are aware of such requirement please point me to that.

Reply:
@atiwari9 We assume the specific requirement for `Alert` (with soft buttons) is also applicable for `SubtleAlert` RPC

Reply:
Can you please point me to the requirement for `Alert` and `DialNumber` as you stated, i can check with author of `SubtleAlert` as well then.

Reply:
Req-t for `DialNumber`:
```
Requirements to SDL:
1. SDL must not apply "default timeout for RPCs processing" for BasicCommunication.DialNumber RPC (that is, SDL must always wait for HMI response to BC.DialNumber as long as it takes and not return GENERIC_ERROR to mobile app).

Information:
HMI is expected to always respond to BC.DialNumber back to SDL:
-> SUCCESS – when the user presses ‘call’ button
-> REJECTED – when HMI is busy with some high-priority event
-> ABORTED –
a) when the user has not yet done his choice and some higher-priority event interrupted the DialNumber dialog
b) when the user presses ‘cancel’ button
-> TIMED_OUT – when the DialNumber dialog is closed by HMI-defined default timeout.
-> GENERIC_ERROR – when anything unpredictable happened on HMI
```

Req-t for `Alert`:
```
SUCCESS result:
1. App sends Alert

{<UI-related params WITHOUT softButtons >, <TTS-related params>}: SDL must return SUCCESS in case TTS engine has successfully spoken TTS-related values and HMI has successfully displayed a message with UI-related values and closed it by timeout.

2. App sends Alert{<UI-related params WITHOUT softButtons >}: SDL must return SUCCESS in case HMI has successfully displayed a message with UI-related values and closed it by timeout.

3. App sends Alert{<UI-related params WITHOUT softButtons >, <TTS-related params>}
: SDL must return SUCCESS in case TTS engine has successfully spoken TTS-related values and HMI has successfully displayed a message with UI-related values and closed it by DEFAULT_ACTION or STEAL_FOCUS button press.

4. App sends Alert

{<UI-related params WITHOUT softButtons >}
: SDL must return SUCCESS in case HMI has successfully displayed a message with UI-related values and closed it it by DEFAULT_ACTION or STEAL_FOCUS button press..

5. App sends Alert

{<UI-related params WITH softButtons >, <TTS-related params>}
: SDL must return SUCCESS in case TTS engine has successfully spoken TTS-related values and HMI has successfully displayed a message with UI-related values and closed it by DEFAULT_ACTION or STEAL_FOCUS button press.

6. App sends Alert

{<UI-related params WITH softButtons >}
: SDL must return SUCCESS in case HMI has successfully displayed a message with UI-related values and closed it it by DEFAULT_ACTION or STEAL_FOCUS button press..

Pay attention!
Timeout for Alert dialog WITH SoftButtons must NOT be successful. See ABORTED result code.

ABORTED result (add the following):
1. App sends Alert RPC WITH SoftButtons: SDL must return ABORTED in case HMI has successfully displayed a message with UI-related values and closed either by HMI-defined timeout, or by app-defined requested timeout, or by HU System event of higher priority.
```

Note that case with `softButtons` is a legacy code that is present in the codebase initially:
https://github.com/smartdevicelink/sdl_core/blob/60e3354b17bcecd20d7727c71a58b34ba81f4af0/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/alert_request.cc#L85